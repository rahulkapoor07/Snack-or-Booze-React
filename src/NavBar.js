import React from "react";
import "./NavBar.css";
import { Link } from "react-router-dom";
import { Navbar, Nav, NavItem } from "reactstrap";

function NavBar() {
  return (
    <div>
      <Navbar expand="md">
        <Link to="/" className="navbar-brand">
          Snack or Booze
        </Link>

        <Nav className="ml-auto" navbar>
          <NavItem>
            <Link to="/snacks">Snacks</Link>
          </NavItem>
          <NavItem>
          <Link to="/drinks">Drinks</Link>
          </NavItem>
          <NavItem>
          <Link to="/add-new">Add New</Link>
          </NavItem>
        </Nav>
      </Navbar>
    </div>
  );
}

export default NavBar;
