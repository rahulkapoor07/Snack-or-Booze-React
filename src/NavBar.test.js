import React from "react";
import {MemoryRouter} from "react-router-dom";
import { render } from '@testing-library/react';
import NavBar from './NavBar';


describe("Testing Navbar component", ()=>{
test("render", ()=>{
    render(<MemoryRouter><NavBar /></MemoryRouter>)
});

test("SnapShot test", ()=>{
    const {asFragment}= render(<MemoryRouter><NavBar /></MemoryRouter>);
    expect(asFragment()).toMatchSnapshot();
});
});