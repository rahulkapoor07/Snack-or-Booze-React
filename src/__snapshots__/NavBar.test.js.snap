// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Cafe component SnapShot test 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Snack or Booze
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/snacks"
          >
            Snacks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/drinks"
          >
            Drinks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/add-new"
          >
            Add New
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Testing Navbar component SnapShot test 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Snack or Booze
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/snacks"
          >
            Snacks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/drinks"
          >
            Drinks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/add-new"
          >
            Add New
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;

exports[`Testing Navbar component checking text 1`] = `
<DocumentFragment>
  <div>
    <nav
      class="navbar navbar-expand-md"
    >
      <a
        aria-current="page"
        class="navbar-brand active"
        href="/"
      >
        Snack or Booze
      </a>
      <ul
        class="ml-auto navbar-nav"
      >
        <li
          class="nav-item"
        >
          <a
            href="/snacks"
          >
            Snacks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/drinks"
          >
            Drinks
          </a>
        </li>
        <li
          class="nav-item"
        >
          <a
            href="/add-new"
          >
            Add New
          </a>
        </li>
      </ul>
    </nav>
  </div>
</DocumentFragment>
`;
